### ISSUE TITLE
Start arana server, execute TestSelect method in integration_test.go will case panic.

### ISSUE THREAD

**Issue by dongzl** [dubbogo/arana#34]
**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

```
2021-12-30T16:22:00.085+0800    DEBUG   executor/redirect.go:137        ComQuery: SET NAMES utf8mb4
2021-12-30T16:30:24.726+0800    DEBUG   executor/redirect.go:137        ComQuery: SET NAMES utf8mb4
^[[20~2021-12-30T16:30:37.444+0800      DEBUG   executor/redirect.go:137        ComQuery: SET NAMES utf8mb4
2021-12-30T16:30:46.986+0800    DEBUG   executor/redirect.go:137        ComQuery: SET NAMES utf8mb4
2021-12-30T16:31:01.774+0800    DEBUG   executor/redirect.go:137        ComQuery: SET NAMES utf8mb4
2021-12-30T16:31:04.551+0800    DEBUG   executor/redirect.go:137        ComQuery: SET NAMES utf8mb4
2021-12-30T16:31:04.553+0800    INFO    executor/redirect.go:193        SELECT emp_no, birth_date, first_name, last_name, gender, hire_date FROM employees 
                WHERE emp_no = 100001
2021-12-30T16:31:04.554+0800    ERROR   mysql/server.go:156     mysql_server caught panic:
trying to call recycleReadPacket while currentEphemeralPolicy is 0
github.com/dubbogo/arana/pkg/mysql.(*Listener).handle.func1
        /Users/dongzonglei/source_code/Github/arana/pkg/mysql/server.go:156
runtime.gopanic
        /usr/local/go/src/runtime/panic.go:971
github.com/dubbogo/arana/pkg/mysql.(*Conn).recycleReadPacket
        /Users/dongzonglei/source_code/Github/arana/pkg/mysql/conn.go:345
github.com/dubbogo/arana/pkg/mysql.(*Listener).handle
        /Users/dongzonglei/source_code/Github/arana/pkg/mysql/server.go:186
```

**Comment by Lvnszn** [comment]
When I start arana firset then start mysql container. Then reproduce it.
- start arana with cmd.
- docker-compose up -d mysql
- testSelect

The console of server show as below
![image](https://user-images.githubusercontent.com/12541438/147749322-2426650d-25f2-49ef-ad42-48d0826426f8.png)

The console of test show as below
![image](https://user-images.githubusercontent.com/12541438/147749314-8fed6bdc-2547-4284-b782-cdc158b462bc.png)

**Comment by dk-lockdown** [comment]
need ComQuit command, before connection close.

**Comment by Lvnszn** [comment]
@dongzl 
这个 recycleRead，之前一定有个readEphemeralPacketDirect，这个方法，这两个是成对出现的，readEphemeralPacketDirect 将currentEphemeralPolicy 设置为 ephemeralRead，recycelRead 将 currentEphemeralPolicy 设置为 ephemeralUnused，前提是 currentEphemeralPolicy = ephemeralRead，否则报错。

重点检查这两个方法是不是成对出现，是否有地方调用了 recycleRead，而没有在之前调用 readEphemeralPacketDirect

**Comment by dongzl** [comment]
Fixes by #79 .
